var/spack/repos/builtin/packages/gpgme/package.py

```diff
@@ -33,7 +33,7 @@ class Gpgme(AutotoolsPackage):
 
     depends_on("gnupg", type="build")
     depends_on("libgpg-error", type="build")
-    depends_on("libassuan", type="build")
+    depends_on("libassuan@2.4.2:2", type="build")
```

Comment:
I'm getting build errors (see below) unless I use the following:
```suggestion
    depends_on("libassuan@2.4.2:2", type=("build", "link"))
```
Here is what I get on github codespaces on this branch with `spack install --fresh gpgme`:
```console
     447    libtool: link: /workspaces/spack/lib/spack/env/gcc/gcc -I/workspaces/spack/opt/spack/linux-ubuntu20.04-x86_64_v3/gcc-11.4.0/libassuan-2.5.7-alcdo4rlhyjkzcm467eo2wgdwhm2rs
            ik/include -I/workspaces/spack/opt/spack/linux-ubuntu20.04-x86_64_v3/gcc-11.4.0/libgpg-error-1.50-pv2l5yvh672tnnqzlged354lf3sw3da7/include -g -O2 -Wall -Wcast-align -Wsha
            dow -Wstrict-prototypes -Wno-format-y2k -Wno-missing-field-initializers -Wno-sign-compare -Wno-format-zero-length -Wno-format-truncation -Wno-sizeof-pointer-div -o .libs/
            gpgme-json gpgme-json.o cJSON.o  -lm ./.libs/libgpgme.so -L/workspaces/spack/opt/spack/linux-ubuntu20.04-x86_64_v3/gcc-11.4.0/libgpg-error-1.50-pv2l5yvh672tnnqzlged354lf3
            sw3da7/lib -lgpg-error -Wl,-rpath -Wl,/workspaces/spack/opt/spack/linux-ubuntu20.04-x86_64_v3/gcc-11.4.0/gpgme-1.20.0-id6xov5iqbe3k4xozgn3xmdomclmkxlh/lib
     448    /usr/bin/ld: warning: libassuan.so.0, needed by ./.libs/libgpgme.so, not found (try using -rpath or -rpath-link)
  >> 449    /usr/bin/ld: ./.libs/libgpgme.so: undefined reference to `assuan_send_data@@LIBASSUAN_1.0'
  >> 450    /usr/bin/ld: ./.libs/libgpgme.so: undefined reference to `assuan_set_flag@@LIBASSUAN_1.0'
  >> 451    /usr/bin/ld: ./.libs/libgpgme.so: undefined reference to `assuan_ctx_set_system_hooks@@LIBASSUAN_1.0'
  >> 452    /usr/bin/ld: ./.libs/libgpgme.so: undefined reference to `assuan_release@@LIBASSUAN_1.0'
  >> 453    /usr/bin/ld: ./.libs/libgpgme.so: undefined reference to `assuan_transact@@LIBASSUAN_1.0'
  >> 454    /usr/bin/ld: ./.libs/libgpgme.so: undefined reference to `assuan_write_line@@LIBASSUAN_1.0'
  >> 455    /usr/bin/ld: ./.libs/libgpgme.so: undefined reference to `assuan_read_line@@LIBASSUAN_1.0'
  >> 456    /usr/bin/ld: ./.libs/libgpgme.so: undefined reference to `assuan_socket_connect@@LIBASSUAN_1.0'
  >> 457    /usr/bin/ld: ./.libs/libgpgme.so: undefined reference to `assuan_pending_line@@LIBASSUAN_1.0'
  >> 458    /usr/bin/ld: ./.libs/libgpgme.so: undefined reference to `assuan_sendfd@@LIBASSUAN_1.0'
  >> 459    /usr/bin/ld: ./.libs/libgpgme.so: undefined reference to `assuan_pipe_connect@@LIBASSUAN_1.0'
  >> 460    /usr/bin/ld: ./.libs/libgpgme.so: undefined reference to `assuan_get_active_fds@@LIBASSUAN_1.0'
  >> 461    /usr/bin/ld: ./.libs/libgpgme.so: undefined reference to `assuan_new_ext@@LIBASSUAN_1.0'
  >> 462    collect2: error: ld returned 1 exit status
  >> 463    make[1]: *** [Makefile:852: gpgme-json] Error 1
```

Reply:
(Build succeeds fine with the suggested change.)